Mage.Sets/src/mage/cards/g/GrenzosRebuttal.java

```diff
@@ -0,0 +1,122 @@
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{4}{R}{R}");
+
+        // Put a 4/4 red Ogre creature token onto the battlefield. Starting with you, each player chooses an artifact, a creature, and a land from among the permanents controlled by the player to their left. Destroy each permanent chosen this way.
+        this.getSpellAbility().addEffect(new CreateTokenEffect(new Ogre44Token()));
+        this.getSpellAbility().addEffect(new GrenzosRebuttalEffect());
+    }
+
+    private GrenzosRebuttal(final GrenzosRebuttal card) {
+        super(card);
+    }
+
+    @Override
+    public GrenzosRebuttal copy() {
+        return new GrenzosRebuttal(this);
+    }
+}
+
+class GrenzosRebuttalEffect extends OneShotEffect {
+
+    private static final List<CardType> cardTypes = Arrays.asList(
+            CardType.ARTIFACT, CardType.CREATURE, CardType.LAND
+    );
+
+    GrenzosRebuttalEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "Starting with you, each player chooses an artifact, a creature, and a land from among the permanents controlled by the player to their left. " + "Destroy each permanent chosen this way.";
+    }
+
+    private GrenzosRebuttalEffect(final GrenzosRebuttalEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public GrenzosRebuttalEffect copy() {
+        return new GrenzosRebuttalEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        UUID controllerId = source.getControllerId();
+        Player controller = game.getPlayer(controllerId);
+        if (controller == null || !controller.canRespond()) {
+            return false;
+        }
+        List<Permanent> chosenPermanents = new ArrayList<>();
+
+        PlayerList playerList = game.getState().getPlayersInRange(controllerId, game, true);
+
+        Player currentPlayer = game.getPlayer(playerList.get());
+        Player nextPlayer = null;
+
+        while (!playerList.get().equals(controllerId) || nextPlayer == null) {
+            nextPlayer = game.getPlayer(playerList.getNext());
+            if (nextPlayer == null) {
+                return false;
+            }
+
+            if (currentPlayer != null) {
+                for (CardType cardType : cardTypes) {
+                    String cardTypeText = CardUtil.addArticle(cardType.toString());
+                    FilterPermanent filter = new FilterPermanent(
+                            cardTypeText + " controlled by " + nextPlayer.getLogName());
+                    filter.add(cardType.getPredicate());
+                    filter.add(new ControllerIdPredicate(nextPlayer.getId()));
+
+                    Target target = new TargetPermanent(1, 1, filter, true);
+
+                    if (target.canChoose(currentPlayer.getId(), source, game)) {
+                        while (currentPlayer.canRespond() && !target.isChosen(game) && target.canChoose(currentPlayer.getId(), source, game)) {
```

Comment:
Do not run such while cycle. Call single chooseTarget and get target result. That’s all. Game engine will process all other use cases (e.g. allow to continue on no valid targets, on disconnected, etc).

P.S.
AI don’t like while code. If you really need infinite while choices (not in that PR) then remove selected id from available list. So AI will finish it in any use cases instead freeze in a selection of the same non-valid target.